developer/debugging/automated-tests.md

```diff
@@ -17,23 +17,31 @@ Starting with Qubes R3 we use [python unittest][unittest] to perform automatic t
 Despite the name, we use it for both [unit tests](https://en.wikipedia.org/wiki/Unit_tests) and [integration tests](https://en.wikipedia.org/wiki/Integration_tests). 
 The main purpose is, of course, to deliver much more stable releases.
 
-Integration tests are written with the assumption that they will be called on dedicated hardware. 
-**Do not run these tests on installations with important data, because you might lose it.**
-Since these tests were written with this expectation, all the VMs with a name starting with `test-` on the installation are removed during the process, and all the tests are recklessly started from dom0, even when testing VM components.
-
-Most of the tests are stored in the [core-admin repository](https://github.com/QubesOS/qubes-core-admin/tree/master/qubes/tests) in the `qubes/tests` directory. 
-To start them you can use the standard python unittest runner:
+The tests are written with the assumption that they will be called on dedicated hardware.
 
-`python3 -m unittest -v qubes.tests`
-    
-Or our custom one:
+**Do not run these tests on installations with important data, because you might lose it.**
 
-`python3 -m qubes.tests.run -v`
+Since the tests were written with this expectation, all the VMs with a name starting with `test-` on the installation are removed during the process, and all the tests are recklessly started from dom0, even when testing VM components.
 
-Our test runner runs mostly the same as the standard one, but it has some nice additional features like color output and not needing the "qubes.test" prefix. 
-It also has the ability to run lone selected template tests.
+Most of the tests are stored in the [core-admin repository](https://github.com/QubesOS/qubes-core-admin/tree/master/qubes/tests) in the `qubes/tests` directory. 
 
-You can use `python3 -m qubes.tests.run -h` to get usage information:
+To run tests, you need to do the following steps:
+1. `git clone https://github.com/QubesOS/qubes-core-qrexec` and put it to `/root/` in dom0 on your test machine.
+2. `git clone https://github.com/QubesOS/qubes-core-admin` and put it to `/root/` in dom0 on your test machine.
+3. Inside `dom0` on your test machine, create a script at `/root/qubes-core-qrexec/run`:
+    ```{.bash}
+    #!/bin/bash
+    [[ "$(whoami)" == "root" ]] || { >&2 echo "This script must be run as root." ; exit 1 ; }
+    systemctl stop qubesd
+    export PYTHONPATH=/root/qubes-core-qrexec
+    ./run-tests -v $(python3 -m qubes.tests.run -l | grep "$@")
```

Comment:
The "run-tests" script is a helper script for running tests directly from the repository, not necessary in dom0. This is just for selected unit tests that actually can run outside of dom0.
If you run them in dom0, use `python3 -m qubes.tests.run` directly, as you can see in instructions for Qubes 4.0.
If you like, you can still provide it a list of tests to run (as shows in an example below).